synapse/neuron.py

```diff
@@ -490,11 +491,10 @@ def onlink(ok, link):
 
             isok, sess = retn.wait(timeout=timeout)
             if not isok:
-                # XXX Raise / log here?
-                return None
+                raise s_common.RetnTimeout(msg='retnwait timed out or failed')
 
             if not sess.waittx(timeout=timeout):
-                return None
+                raise s_common.RetnTimeout(msg='waittx timed out or failed')
```

Comment:
i actually think that at this point we could leave the context manager (fini' the retnwait object) then do the sess.waittx(). Whether we raise or return None here is a different question; but raising RetnTimeout here doesn't make sense.